### PR TITLE
lsfd: refine column name and sematics

### PR DESCRIPTION
**Pull request by masatake**
This pull request finishes "homeworks" prepared in #1770.

This pull request does
* split NAME column into NAME and PROCNAME,
* split TYPE column into TYPE and STTYPE, and
* introduce PIDFD.{COMM,PID,NSPID} columns.

NAME and TYPE are talkative.
PROCNAME, STTYPE, and PIDFD.{COMM,PID,NSPID} show raw information suitable for machine processing.

This pull request breaks the compatibility of TYPE column.

    lsfd: use TYPE column to show cooked file types
    
    lsfd has two exclusive requirements：
    
    A. show quickly understandable information in limited display area, and
    B. provide primitive raw information for post-process tools.
    
    For the purpose A, summarizing information is nice. However, the
    summarized information is not suitable for tools.
    
    To satisfy the both requirements, I plan introducing new fields.
        
    STTYPE represents the type information returned from stat(2).
    TYPE is the same as STTYPE but two exceptions; TYPE is filled with
    PROTONAME value if the target fd is a socket, or with AINODECLASS
    value if the target fd is a anonymous inode.
    
    Below an example output:
    
        $ ./lsfd -o COMMAND,TYPE,STTYPE,NAME -Q '(TYPE != STTYPE)'  | head
        COMMAND                TYPE STTYPE NAME
        systemd         UNIX-STREAM   SOCK UNIX-STREAM:[25493]
        systemd         UNIX-STREAM   SOCK UNIX-STREAM:[25493]
        systemd                UNIX   SOCK UNIX:[31445]
        systemd           eventpoll   UNKN anon_inode:[eventpoll]
        systemd            signalfd   UNKN anon_inode:[signalfd]
        systemd             notify    UNKN anon_inode:inotify
        systemd             timerfd   UNKN anon_inode:[timerfd]
        systemd           eventpoll   UNKN anon_inode:[eventpoll]
        systemd             notify   UNKN anon_inode:inotify
    
    STTYPE, column for tools, is printed only when the column is specified
    with -o option.

This pull request also breaks the compatibility of NAME column.

    lsfd: use NAME column to show cooked file names
    
    This change splits the NAME column
    into cooked NAME column and raw PROCNAME column.
    
    PROCNAME represents the type information extracted from /proc/$pid/fd or
    /proc/$pid/map_files. NAME is the same as PROCNAME but humans friendly
    and talkative if a special handler is implemented for the type of the
    file.
    
    Currently only pidfd has such a handler.
    
    Below an example output:
    
        $ ./lsfd -o TYPE,STTYPE,PROCNAME,NAME -Q '(TYPE == "pidfd")'
         TYPE STTYPE PROCNAME           NAME
        pidfd   UNKN anon_inode:[pidfd] pid=4661 comm=dbus-broker nspid=4661
        pidfd   UNKN anon_inode:[pidfd] pid=4924 comm=dbus-broker nspid=4924

This pull request introduces PIDFD.{PID,COMM,NSPID} columns.

    lsfd: show pid, comm, and nspid of pidfd in PIDFD.{PID,COMM,NSPID} columns
    
    The values shown in the new columns are already shown in NAME column.
    Showing these values together in the NAME column helps users
    understand the pidfds opened quickly. However, the showing them
    together is not suitable for machine processing. The newly introduced
    columns are for machine processing.
    
    Below an example output:
    
        $ ./lsfd -o COMMAND,TYPE,PIDFD.COMM,PIDFD.PID -Q '(TYPE == "pidfd")'
        COMMAND          TYPE PIDFD.COMM  PIDFD.PID
        dbus-broker-lau pidfd dbus-broker      4661
        dbus-broker-lau pidfd dbus-broker      4924

### CODE REVIEW

File: tests/ts/lsfd/column-ainodeclass [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# This file is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+TS_TOPDIR="${0%/*}/../.."
+TS_DESC="ainodeclass column"
+export TS_DESC
+
+. "$TS_TOPDIR"/functions.sh
```
[the remark this answers is not in the document]
Comment (masatake) [reply tests/ts/lsfd/column-ainodeclass:21]: Should I update .github/workflows/differential-shellcheck.yml?

File: tests/ts/lsfd/column-type [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# This file is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+TS_TOPDIR="${0%/*}/../.."
+TS_DESC="TYPE and STTYPE column"
+export TS_DESC
+
+. $TS_TOPDIR/functions.sh
```
[the remark this answers is not in the document]
Comment (masatake) [reply tests/ts/lsfd/column-type]: Should I update .github/workflows/differential-shellcheck.yml?

File: tests/ts/lsfd/column-name [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+# This file is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+TS_TOPDIR="${0%/*}/../.."
+TS_DESC="NAME and PROCNAME column"
+export TS_DESC
+
+. "$TS_TOPDIR"/functions.sh
```
[the remark this answers is not in the document]
Comment (masatake) [reply tests/ts/lsfd/column-name:21]: Should I update .github/workflows/differential-shellcheck.yml?

**Comment by karelzak** [comment]
We do not use "export TS_DESC" in other tests; why do we need it now? (And ignore static analyzers if you think you're right :-))

It's good that you have removed the '_' prefix in column names (_TYPE); for example, STTYPE is a good name for the column. But I'm not sure about PROCNAME; it sounds like a process name; what about FDNAME (it's from /proc/#/fd) or KNAME (it's from the kernel), or FDPATH, FDREF (reference), or so?

File: tests/ts/lsfd/column-ainodeclass (hunk 1 above)
[the remark this answers is not in the document]
Comment (jamacku) [reply tests/ts/lsfd/column-ainodeclass:20]: Sorry for the inconvenience. This issue should be resolved via the following PR:

- https://github.com/util-linux/util-linux/pull/1833

**Pull request merged**